documentation/writing_tasks.md

```diff
@@ -1077,3 +1077,453 @@ path.
 For more information on securely passing user input, see the blog post [Stop
 using backtick to run shell command in
 Ruby](https://www.hilman.io/blog/2016/01/stop-using-backtick-to-run-shell-command-in-ruby/).
+
+## Debugging tasks
+
+Because tasks are executed remotely, it can be challenging to debug tasks that
+are not working as expected. There are several ways that you can debug tasks,
+including using remote debugging libraries, using methods available in the
+task helper libraries, running the task locally as a script, and redirecting
+`stderr` to `stdout`.
+
+### Debug logs
+
+Typically, Bolt will only display task output that is sent to `stdout`.
+However, Bolt does log additional information about a task run, including
+output sent to `stderr`, at the `debug` level. You can view these logs
+during a task run using the `--log-level debug` CLI option.
+
+```shell
+$ bolt task run mytask param1=foo param2=bar -t all --log-level debug
+```
+
+### Debuggers
```

Comment:
This section is a little confusing. It's written as if you're debugging against a remote target, but the Ruby and PS instructions only work for localhost, while the Python instructions do work for a remote target. There's just a "tip" for Ruby/PS that says it's _simpler_ to use localhost, but they don't specify that the instructions are only for localhost.

Reply:
Ya, this one in particular is tricky since connecting to the sockets that are opened by the Ruby/Python debuggers can be difficult. I think the benefit of running the task on localhost is that you don't need to make changes to the task to run it as a script -- I can make that clearer / less confusing in this section.

For PowerShell, James' suggestion was to always debug it as a script, since remote debugging with PowerShell can be really complicated. I can move this section under "Running a task as a script", but it seems more appropriate to have it under "Debuggers".